### PR TITLE
refactor: pass http method to metasrv http handler

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Spotted on https://github.com/GreptimeTeam/docs/pull/878.

This follows up #3527 

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/meta-srv/src/service/admin/maintenance.rs [hunk 1]
```diff
@@ -91,13 +92,17 @@ impl MaintenanceHandler {
...
+        method: http::Method,
         params: &HashMap<String, String>,
     ) -> crate::Result<Response<String>> {
-        if path.ends_with("/set") {
-            self.set_maintenance(params).await
-        } else {
-            self.get_maintenance().await
+        match method {
+            http::Method::GET => self.get_maintenance().await,
+            http::Method::PUT => self.set_maintenance(params).await,
```
Comment (nicecui) [inline src/meta-srv/src/service/admin/maintenance.rs:101]: In general, the PUT method sends data via the request payload. Are the params coming from the request payload?

Comment (tisonkun) [reply src/meta-srv/src/service/admin/maintenance.rs:101]: No. We're using query params still. Pass and decode body can be quite complicated. I tend not to implement it in this PR.

**Review by nicecui — APPROVED** [review]

File: src/meta-srv/src/service/admin.rs [hunk 2]
```diff
@@ -163,7 +163,8 @@ where
             })
             .unwrap_or_default();
         let path = req.uri().path().to_owned();
-        Box::pin(async move { router.call(&path, query_params).await })
+        let method = req.method().to_owned();
```
Comment (killme2008) [inline src/meta-srv/src/service/admin.rs:166]: Prefer to use `clone()`.
```suggestion
        let method = req.method().clone();
```

**Review by killme2008 — APPROVED** [review]
LGTM. @fengjiachun PTAL

File: src/meta-srv/src/service/admin.rs (hunk 2 above)
Comment (tisonkun) [reply src/meta-srv/src/service/admin.rs]: Updated.

**Pull request merged**